src/structlog/_base.py

```diff
@@ -113,39 +97,42 @@ def try_unbind(self, *keys):
 
         return bl
 
-    def new(self, **new_values):
+    def new(self, **new_values: Any) -> "BoundLoggerBase":
         """
-        Clear context and binds *initial_values* using :func:`bind`.
+        Clear context and binds *initial_values* using `bind`.
 
         Only necessary with dict implementations that keep global state like
         those wrapped by `structlog.threadlocal.wrap_dict` when threads
         are re-used.
-
-        :rtype: ``self.__class__``
         """
         self._context.clear()
+
         return self.bind(**new_values)
 
     # Helper methods for sub-classing concrete BoundLoggers.
 
-    def _process_event(self, method_name, event, event_kw):
+    def _process_event(
+        self, method_name: str, event: Optional[str], event_kw: Dict[str, Any]
+    ) -> Tuple[Tuple, Mapping]:
```

Comment:
I usually prefer to add [`disallow_any_generics`](https://github.com/pre-commit/pre-commit/blob/8d28c9ab40aec25236ba080fb7202b0a279b09d4/setup.cfg#L62) so that mypy can better understand the types (instead of decaying to `Any` in a bunch of places)

without typing these you get `Tuple[Tuple[Any, ...], Mapping[Any, Any]]` (which even if that's true I think is better / more explicit than the bare un-generic'd types)

Reply:
Ah nice that's a bunch new errors for me to deal with tomorrow. ;)